MakingItBetterG55/src/application/TextRetriever.java

```diff
@@ -0,0 +1,70 @@
+package application;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextRetriever {
+    //path to the CSV containing all text in the game
+    private static final String CSV_PATH = "MakingItBetterG55/src/making-it-better-text.csv";
+
+    //list containing all text from the CSV
+    private List<String[]> translations;
+
+    //index representing current language
+    private int currentLanguageIndex;
+
+    //class is initialised by converting the CSV values into a List and currentLanguageIndex set to 0
+    public TextRetriever(){
+        this.translations = readTranslationsFromCSV(CSV_PATH);
+        this.currentLanguageIndex = 0;
+    }
+
+    //retrieves text based on provided index and language index
+    public String getText(int index, int languageIndex){
+        if(index >= 0 && index < translations.size()){
+            if(languageIndex >= 0 && languageIndex < translations.get(0).length){
+                return translations.get(index)[languageIndex];
+            } else {
+                return "Invalid Text";
+            }
+        } else {
+            return "Invalid text";
+        }
+    }
+
+    //changes the current language index
+    public void setLanguage(int languageIndex){
```

Comment:
Right now this doesn't really have a use case since the the language actually used is given in the argument of getText, I would probably do one of three things: remove this method and the current language index class variable, change getText to only take an index and use whatever the set language by this function is, or use some method overloading to have a second version of getText which only takes an index so you can decide whether you want to explicitly say which language or just use whichever it is set as.

Reply:
I included it just so there would be a way to easily change the language after it's already set but you're right, it's unlikely to have a lot of uses. I'll take a look tomorrow and make changes to it. Thanks for the feedback!

Reply:
Of course, and yeah that would be cool functionality to add, but as it stands this wouldn't update already existing text to a new language which you could do but probably more trouble than its worth needing to mess around with Listeners and stuff, so in my opinion, it will be better to just reload the page contents when the language is changed for a project like this.

Reply:
You're right, it's a tad more complicated than I'd thought. Thanks!